### DISCUSSION TITLE
Live embed

### DISCUSSION THREAD

**Discussion by datdinhquoc** [jgraph/drawio#2073] [Ideas]
Drawio is perfect for drawing diagrams, it can generate iframe code to embed to websites. 

However, the embed is not live embed, which means when the diagram is edited, the embed doesn't change at all. I can see something like data in the embed code.

Should the embed point to some URL instead of static data? so the embed is a live embed, and changes after the original diagram is edited.

**Comment by z4ch523** [comment]
This can be achieved when you store your diagram in google drive. It will change the content of the iframe into a reference to this file.

**Comment by alderg** [comment]
Where supported, we will try and use a link to the public diagram file. If the cloud storage does not support such links (eg. in OneDrive), we use a copy of the diagram. To use a link instead, replace  the src attribute of the iframe with the URL of the editor, replacing https://app.diagrams.net with https://viewer.diagrams.net, eg.
`<iframe frameborder="0" style="width:100%;height:345px;" src="https://viewer.diagrams.net/#W..."></iframe>`.
However, this will require the user to login to view the diagram.

If your file was saved as `.drawio.svg` or `.drawio.png`, you can use a direct link to the image in an `img` tag for viewing the diagram. To obtain the link, use the `Raw` button or a link of the form _raw.githubusercontent.com/${user}/${repo}/${branch}/${path}_ for public GitHub repositories. For OneDrive, the _</> Embed_ button seems to produce an image link in some cases, but it is unclear under what conditions it creates an `iframe` instead.
